core/shared/src/main/scala/zio/prelude/ZValidation.scala

```diff
@@ -104,6 +104,14 @@ sealed trait ZValidation[+W, +E, +A] { self =>
       case Success(w, _) => w
     }
 
+  /**
+   * Returns the value, if successful, or the transformed (using `f`) failure.
+   */
+  final def getOrElse[A1 >: A](f: Failure[W, E] => A1): A1 = this match {
```

Comment:
Maybe one variant that just gives you the errors and another `getOrElseLog` variant that gives you a tuple of the log and the errors?

Reply:
With `Failure`, you may get your hands on `message`, which is my point here :smile_cat: 

But I can add another method that will take just `f: NonEmptyChunk[E] => A1`. Do you think, it would be useful?

Reply:
It just seems a little irregular to expose one of the constructors in a fold variant as opposed to the underlying data values, at least where it isn't a fixed point type. Maybe a `prettyPrint` or `render` method on `ZValidation` could address this use case?

Reply:
What I'm trying to achieve here, is to make it super easy to do something like this:
```scala
.getOrElse(e => sys.error(e.message)) // this whole thing returns the successful value (or throws)
```
So I need to have the member `message` on `Failure`.

Reply:
Could we put it on `ZValidation` itself so it is like `validation.getOrElse(_ => sys.error(validation.render))`? Definitely want to make it easy to do something like that. Just having separate operators on the subtype of an algebraic data type and returning the subtype instead of its contents in a fold seems like a bit of a code smell.

Reply:
But then you have to use the variable (`validation` in your example) twice. In a situation, when you're juggling with multiple values of `Validation`, you have to make special care that you use the right one at the right place -- that's an unnecessary room for errors.

What I'm trying to replicate is an analogue of `valueOr` from cats:
https://typelevel.org/cats/api/cats/syntax/EitherOps.html#valueOr[BB%3E:B](f:A=%3EBB):BB

I've renamed it, so that it better reflects the meaning, `getOrElse` is already established for a slightly different concept.